pkgs/by-name/ma/mattermost/package.nix

```diff
@@ -1,88 +1,135 @@
-, fetchurl
-, nixosTests
+{
+  lib,
+  callPackage,
+  buildGoModule,
+  fetchFromGitHub,
+  nix-update-script,
+  fetchurl,
+  runtimeShell,
+  nixosTests,
 }:
 
-buildGoModule rec {
-  pname = "mattermost";
-  # ESR releases only.
-  # See https://docs.mattermost.com/upgrade/extended-support-release.html
-  # When a new ESR version is available (e.g. 8.1.x -> 9.5.x), update
-  # the version regex in passthru.updateScript as well.
-  version = "9.11.3";
-
-  src = fetchFromGitHub {
-    owner = "mattermost";
-    repo = "mattermost";
-    rev = "v${version}";
-    hash = "sha256-CuFkydl1ZZUAWmrDIV1Jp9S6jIKYmglAe5XW2lTRgtQ=";
-  };
+let
+  mattermost = buildGoModule rec {
+    pname = "mattermost";
+    # ESR releases only.
+    # See https://docs.mattermost.com/upgrade/extended-support-release.html
+    # When a new ESR version is available (e.g. 8.1.x -> 9.5.x), update
+    # the version regex in passthru.updateScript as well.
+    version = "9.11.3";
 
-  # Needed because buildGoModule does not support go workspaces yet.
-  # We use go 1.22's workspace vendor command, which is not yet available
-  # in the default version of go used in nixpkgs, nor is it used by upstream:
-  # https://github.com/mattermost/mattermost/issues/26221#issuecomment-1945351597
-  overrideModAttrs = (_: {
-    buildPhase = ''
-      make setup-go-work
-      go work vendor -e
+    src = fetchFromGitHub {
+      owner = "mattermost";
+      repo = "mattermost";
+      rev = "v${version}";
+      hash = "sha256-CuFkydl1ZZUAWmrDIV1Jp9S6jIKYmglAe5XW2lTRgtQ=";
+    };
+
+    postPatch = ''
+      # Just echo install/get/mod commands in the Makefile, since the dependencies are locked.
+      substituteInPlace server/Makefile \
+        --replace-warn '$(GO) install' '@echo $(GO) install' \
+        --replace-warn '$(GO) get' '@echo $(GO) get' \
+        --replace-warn '$(GO) get' '@echo $(GO) mod'
+
+      # mmctl tests shell out by writing a bash script to a tempfile
+      substituteInPlace server/cmd/mmctl/commands/config_e2e_test.go \
+        --replace-fail '#!/bin/bash' '#!${runtimeShell}'
     '';
-  });
 
-  webapp = fetchurl {
-    url = "https://releases.mattermost.com/${version}/mattermost-${version}-linux-amd64.tar.gz";
-    hash = "sha256-4JzhL2+G3T98pNFgKugs/eoSrbm7QSk5grVlprrIKEI=";
-  };
+    # Needed because buildGoModule does not support go workspaces yet.
+    # We use go 1.22's workspace vendor command, which is not yet available
+    # in the default version of go used in nixpkgs, nor is it used by upstream:
+    # https://github.com/mattermost/mattermost/issues/26221#issuecomment-1945351597
+    overrideModAttrs = (
+      _: {
+        buildPhase = ''
+          make setup-go-work
+          go work vendor -e -v
+        '';
+      }
+    );
 
-  # Makes nix-update-script pick up the fetchurl for the webapp.
-  # https://github.com/Mic92/nix-update/blob/1.3.1/nix_update/eval.py#L179
-  offlineCache = webapp;
+    webapp = fetchurl {
```

Comment:
Coming for this next...

Reply:
Sadly does not appear to be possible since the lockfile is inconsistent with hashes.